types/yup/index.d.ts

```diff
@@ -100,6 +101,10 @@ export interface MixedSchemaConstructor {
 // tslint:disable-next-line:no-empty-interface
 export interface MixedSchema extends Schema<any> {}
 
+export interface MixedSchemaLocale extends MixedSchema {
+    notType(ref?: { path: string, type: string, value: any, originalValue: any }): string;
```

Comment:
Is it worthwhile to add the rest of the mixed schema locale properties as string to this object? I've only ever used yup with my own localization around it; I'm not sure if it's worthwhile so I'm just trying to generate context!